### PR TITLE
Add cgsi-gsoap

### PR DESCRIPTION
**Pull request by chrisburr**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/cgsi-gsoap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - make
+    - pkg-config
+  host:
+    - gsoap =2.7
+    - voms
+    - gct
+    - openssl 1.0.2r
```
Comment (chrisburr) [inline recipes/cgsi-gsoap/meta.yaml]: I'm using openssl 1.0 as I'm building something which needs it so I'll migrate over to 1.1 in the feedstock.

I've checked and `run_export` is used correctly in `openssl=1.0.2r`.

**Comment by chrisburr** [comment]
@conda-forge/staged-recipes This is ready for review.

File: recipes/cgsi-gsoap/build.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+fi
+
+cmake -LAH \
+    "${CMAKE_PLATFORM_FLAGS[@]}" \
+    -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
+    -DLIB_SUFFIX="" \
+    ..
+
+make -j${CPU_COUNT}
+make install
```
Comment (scopatz) [inline recipes/cgsi-gsoap/build.sh]: Is there a make check?

Comment (chrisburr) [reply recipes/cgsi-gsoap/build.sh]: No there isn't

**Pull request merged**